File: 154-FindMinimumInRotatedSortedArrayIi/154-FindMinimumInRotatedSortedArrayIi.py
# Last updated: 7/14/2026, 2:18:23 PM
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0 
        r = len(nums)-1
        mid = 0
        while l<r:
            mid = l + (r-l)//2
            if nums[mid]>nums[r]:
                l = mid+1
            elif nums[mid]<nums[r]:
                r = mid
            else:
                r-=1
        return nums[l]

File: 154-FindMinimumInRotatedSortedArrayIi/154-FindMinimumInRotatedSortedArrayIi.java
// Last updated: 7/14/2026, 2:18:28 PM
import java.util.*;
class Solution {
    public int findMin(int[] nums) {
        // int a =Arrays.stream(nums).min().getAsInt();
        // return a;
        int temp = nums[0];
        for(int i=1;i<nums.length;i++){
            if(temp>nums[i]){
                temp=nums[i];
            }
        }
        return temp;
    }
}